### PR TITLE
Breadcrumb message param should be optional

### PR DESCRIPTION
**Pull request by jesse-spevack**
There are cases in charity api where we call `breadcrumbs.record` but only provide a data param. This pr adds support for nil message params on the breadcrumbs method to support these cases. See: https://github.com/givelively/charity-api/blob/2cd532df6f374efeab4c52aeb20069eac04c593f/app/models/csv_export.rb#L26 for an example.

### CODE REVIEW

File: lib/exception_notifier.rb [hunk 1]
```diff
@@ -43,12 +43,12 @@ def add_context(type, context)
 
     # @params data [Hash] the key values to add to the breadcrumb
     # @params message [String] the message string to add to the breadcrumb
-    def breadcrumbs(data:, message:)
+    def breadcrumbs(data:, message: nil)
       raise ArgumentError.new('data must be a hash') unless data.is_a?(Hash)
-      raise ArgumentError.new('message must be a string') unless message.is_a?(String)
+      raise ArgumentError.new('when providing a message, it must be a string') if !message.nil? && !message.is_a?(String)
```
Comment (timlawrenz) [inline lib/exception_notifier.rb:48]:
```suggestion
      raise ArgumentError.new('when providing a message, it must be a string') if message.present? && !message.is_a?(String)
```

File: lib/exception_notifier.rb [hunk 2]
```diff
@@ -43,12 +43,12 @@ def add_context(type, context)
...
 
       error_client.breadcrumbs.record do |crumb|
-        crumb.message = message
+        crumb.message = message if message
```
Comment (timlawrenz) [inline lib/exception_notifier.rb:51]:
```suggestion
        crumb.message = message if message.present?
```

**Review by timlawrenz — CHANGES_REQUESTED** [review]
minor smartassery. let me know if they don't make sense.

**Comment by jesse-spevack** [comment]
double negatives hurt my brain. Good call.

File: lib/exception_notifier.rb [hunk 3]
```diff
@@ -43,12 +43,12 @@ def add_context(type, context)
...
     # @params message [String] the message string to add to the breadcrumb
-    def breadcrumbs(data:, message:)
+    def breadcrumbs(data:, message: nil)
       raise ArgumentError.new('data must be a hash') unless data.is_a?(Hash)
-      raise ArgumentError.new('message must be a string') unless message.is_a?(String)
+      raise ArgumentError.new('when providing a message, it must be a string') if message.present? && !message.is_a?(String)
 
       error_client.breadcrumbs.record do |crumb|
-        crumb.message = message
+        crumb.message = message if message.present?
```
Comment (timlawrenz) [inline lib/exception_notifier.rb:51]: sorry, but do we care at all? If message is nil, crumb.message = nil. I think we can just skip the `if`

**Comment by jesse-spevack** [comment]
Let the message be nil!

**Review by timlawrenz — APPROVED** [review]
yay 🎊

**Pull request merged**